datastore/storagebackend/postgresql/gettsattrgroups.go

```diff
@@ -261,64 +270,71 @@ func getTSAttrGroupsIncInstances(
-// Returns nil upon success, otherwise error.
-func getTSAttrGroupsComboOnly(db *sql.DB, cols []string, groups *[]*datastore.TSMdataGroup) error {
+// getTSAttrGroupsComboOnly creates an array of groups from cols such that each group contains a
+// single, unique combination of database values corresponding to cols. Other attributes than those
+// in cols have the default value for the type (i.e. "" for string, etc.).
+//
+// NOTE: cols is assumed to be sanitized by validateAttrs, so there is no risk of SQL injection
+// in the below query.
+//
+// Returns (array of groups, nil) upon success, otherwise (..., error).
+func getTSAttrGroupsComboOnly(db *sql.DB, cols []string) ([]*datastore.TSMdataGroup, error) {
+
 	// query database for unique combinations of cols in time_series, ordered by cols
 	colsS := strings.Join(cols, ",")
 	query := fmt.Sprintf("SELECT DISTINCT %s FROM time_series ORDER BY %s", colsS, colsS)
 	rows, err := db.Query(query)
 	if err != nil {
-		return fmt.Errorf("db.Query() failed: %v", err)
+		return nil, fmt.Errorf("db.Query() failed: %v", err)
 	}
 	defer rows.Close()
 
+	groups := []*datastore.TSMdataGroup{}
+
 	// aggregate rows into groups
 	for rows.Next() {
 		// extract tsMdata from current result row
 		tsMdata, err := scanTsMdata(rows, cols)
 		if err != nil {
-			return fmt.Errorf("scanTsMdata() failed: %v", err)
+			return nil, fmt.Errorf("scanTsMdata() failed: %v", err)
 		}
 
 		// add new group with tsMData as the combo (and leaving the instances
 		// array unset)
-		*groups = append(*groups, &datastore.TSMdataGroup{Combo: tsMdata})
+		groups = append(groups, &datastore.TSMdataGroup{Combo: tsMdata})
 	}
 
-	return nil
+	return groups, nil
 }
 
 // GetTSAttrGroups ... (see documentation in StorageBackend interface)
 func (sbe *PostgreSQL) GetTSAttrGroups(request *datastore.GetTSAGRequest) (
 	*datastore.GetTSAGResponse, error) {
 
-	cols, err := getTSDBColumns(request.Attrs) // get database column names for requested attributes
-	if err != nil {
-		return nil, fmt.Errorf("getTSAttrCols() failed: %v", err)
+	if err := validateAttrs(request.Attrs); err != nil {
+		return nil, fmt.Errorf("validateAttrs() failed: %v", err)
 	}
 
-	groups := []*datastore.TSMdataGroup{}
+	var groups []*datastore.TSMdataGroup
+	var err error
 
 	if request.IncludeInstances {
-		if err := getTSAttrGroupsIncInstances(sbe.Db, cols, &groups); err != nil {
-			return nil, fmt.Errorf("getTSAGroupsIncInstances() failed: %v", err)
+		if groups, err = getTSAttrGroupsIncInstances(sbe.Db, request.Attrs); err != nil {
+			return nil, fmt.Errorf("getTSAttrGroupsIncInstances() failed: %v", err)
 		}
 	} else {
-		if err := getTSAttrGroupsComboOnly(sbe.Db, cols, &groups); err != nil {
-			return nil, fmt.Errorf("getTSAGroupsComboOnly() failed: %v", err)
+		if groups, err = getTSAttrGroupsComboOnly(sbe.Db, request.Attrs); err != nil {
+			return nil, fmt.Errorf("getTSAttrGroupsComboOnly() failed: %v", err)
```

Comment:
Is there a specific reason to define groups and err here instead of using `:=`?

Reply:
Apolgies, didn't pay enough attention. And the return was not visible in the github diff. Exactly why I usually do the diff in the editor, but was in a rush.

Makes perfect sense.